### DISCUSSION TITLE
Vite module federation

### DISCUSSION THREAD

**Discussion by jewe11er** [nrwl/nx#22646] [Feature Requests]
Do you plan to add MF on Vite? It's much faster then WP5
It'll bee greate, if you write article how to migrate from WP5 to Vite for MF

**Comment by FrozenPandaz** [comment]
I believe this is planned.

cc @jaysoo @Coly010

**Comment by nvonbenken** [comment]
Also interested in this.

**Comment by azinod** [comment]
+1

**Comment by jame-earnin** [comment]
https://github.com/module-federation/vite
Module Federation has been supported in module-federation repository.
We are looking for Nx to natively support Module Federation in Vite also.

**Comment by doiya46** [comment]
+1

**Comment by rogier-mars-mck** [comment]
Any updates on this?

Also, would it be reasonable to assume a default webpack host/remote application can be (automatically) migrated to Vite when this feature hits? E.g. using https://nx.dev/packages/vite/generators/configuration

**Comment by sgtsquiggs** [comment]
~is there a roadmap where we can find this feature?~

Roadmap is here: https://github.com/nrwl/nx/discussions/12836

I think we'll need more upvotes/noise to get this feature considered.

**Comment by sgtsquiggs** [comment]
For some color:

NX 16 is working well for us right now. We're using it to build a shared operations portal where different teams are maintaining different sections (front and back ends) with a shared parent frontend/UI shell.

The eventual goal is for separate major sections to be integrated via MF. We are very happy with vite for our frontends and we do not want to introduce webpack unless we absolutely have to.

We can absolutely get module federation working with vite, but the *developer experience in NX is abysmal*. With Webpack, nx knows there are dependencies and how to spin up the parent shell and simple file server (with watcher) so that you can get live previews. Also `affected` works as expected.

With vite we've got to manually define dependencies ourselves. We can "fake" MF by having a dummy that imports the federated module directly but testing, configuration, etc are all outside of NX. It introduces a lot of headache.

**Comment by sgtsquiggs** [comment]
This library would be useful for getting module federation working in a way that is similar to webpack's implementation: https://github.com/originjs/vite-plugin-federation

**Comment by Nesho-Developer** [comment]
Any updates?

**Comment by Evanion** [comment]
I'm also interested in this. I started looking at [Origin.js](https://github.com/originjs/vite-plugin-federation), and the config API seems identical to webpacks. They even support running [mixed webpack/vite bundlers](https://github.com/originjs/vite-plugin-federation?tab=readme-ov-file#integration-with-webpack) between the different hosts and remotes.

**Comment by Coly010** [comment]
There is currently no official support from the Module Federation team for `vite`. 
However, when `vite` switches to `rolldown`, there is a plan for native MF support for Vite.

I know that some people have had success with https://github.com/originjs/vite-plugin-federation

We are still evaluating options currently for official support from Nx.

**Comment by ReinGomezWP** [comment]
Any updates on this? My team is thinking of migrating from VITE to Webpack just so we can use NX MF, but if VITE native support is in the horizon then I'll be great

**Comment by isunilmekala** [comment]
+1 , looking forward for VITE module federation support

**Comment by dhananjaykumar880** [comment]
+1 looking for vite module federation support on build along with storybook

**Comment by v4irajvimu** [comment]
is there any way to do the module federation fro NX vite app?

**Comment by bandman0226** [comment]
Is there any update on this? We have tried using [Originjs](https://github.com/originjs/vite-plugin-federation) with @nx/vite and keep running into issues that cause it to fail builds or not get the remote loaded

**Comment by nxNero** [comment]
is there any updates I was looking forward to use this feature with nx.

**Comment by artnikbrothers** [comment]
Any updates on it? We switched from Webpack to Vite a few years ago, and now we want to start using Nx, but we hate the idea of switching back to Webpack.
